extension/source/Popup/components/Carousel.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import { Mutex } from 'async-mutex';
+
+import TaskQueue from '../../common/TaskQueue';
+import delay from '../../helpers/delay';
+
+type Props = {
+  images: string[];
+};
+
+type State = {
+  imageA: {
+    index: number;
+    position: 'left' | 'middle' | 'right';
+    resetting: boolean;
+  };
+  imageB: {
+    index: number;
+    position: 'left' | 'middle' | 'right';
+    resetting: boolean;
+  };
+};
+
+const positionMap = {
+  left: 71 - 516,
+  middle: 71,
+  right: 71 + 516,
+};
+
+export default class Carousel extends React.Component<Props, State> {
```

Comment:
Should make this a function component. Most of the newer libraries are hook-only and it will be a good idea to follow the new pattern as it has several advantages as well.

For now we can leave the old components as is (as they are cross compatible to an extent)  and refactor them later but for any new one it will be a good idea to switch now.

Reply:
Done 👍